### PR TITLE
Add pull request template

### PR DESCRIPTION
**Pull request by Aschen**
### What's is in this PR ?

- [X] New feature (non-breaking change which adds functionality)

### What's this PR do?

This PR provide a new pull request template to help contributor to write good PR.  
It's not aimed to be a mandatory template but more a general guideline.  
We hope that this kind of well structured template could help the review process.  

### Where should the reviewer start?

 - [pull_request_template.md](https://raw.githubusercontent.com/kuzzleio/kuzzle/add_pull_request_template/.github/pull_request_template.md)

### How should this be manually tested?

  - Step 1:  Add a file named `.github/pull_request_template.md` with markdown to a repo
  - Step 2 : Commit and push on master
  - Step 3 : Open a pull request on your repo

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+### What are the relevant tickets?
+
+<!-- Link issues here -->
+
+### Where should the reviewer start?
+
+<!-- Tell which files are relevant or which parts of the code you are least confident about -->
+<!-- You can link a entire file of you branch like this : https://github.com/<owner>/<project>/blob/<branch-name>/<path-to-the-file> -->
+
+### How should this be manually tested?
```
Comment (scottinet) [inline .github/pull_request_template.md]: I've mixed feelings about that one. I would _personally_ prefer to see that part gone, but I won't battle over it if it stays.

Pro: sometimes it's interesting to help reviewers by describing how they can view the changes by themselves

Con: useless in most cases
* Fixed issues must include unit and/or functional tests, validating the fix (i.e. tests reproducing the problem and failing without the fix)
* New features must include new functional tests
* New features may be complex and thus difficult to detail in that section

Comment (ballinette) [reply .github/pull_request_template.md]: I suggest to maintain this section as an optional one (each PR submitter is free to keep it or not, depending of its pertinence in a specific case)

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+<!--
+  This template is not mandatory.
+  It simply serves to provide a guide to allow a better review of pull requests.
+-->
+### What's is in this PR ?
```
Comment (ballinette) [inline .github/pull_request_template.md]: Suggestion: this section can be merged with `Relevant issues` (see example here: https://github.com/kuzzleio/kuzzle/pull/1093)

File: .github/pull_request_template.md (hunk 1 above)
Comment (scottinet) [reply .github/pull_request_template.md]: fine by me

**Comment by ballinette** [comment]
great job :+1: 
I suggest to distinguish in the comments the required sections ("Changes types", "Description"...) and the optional ones ("How to test", "Screenshots", "Other changes", "Boyscout"...)

File: .github/pull_request_template.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Aschen) [reply .github/pull_request_template.md]: I like `What is in this PR`

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+
+
+- [ ] Change that requires a documentation update
+- [ ] Change that adds new dependencies
+
+### What's this PR do?
```
[the remark this answers is not in the document]
Comment (Aschen) [reply .github/pull_request_template.md]: Same, I prefer the first one. I think that asking question is more natural, you just read the question and answer it.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  Please include a summary of the change and which issue is fixed.
+  Please also include relevant motivation and context.
+  List any dependencies that are required for this change.
+-->
+
+### What are the relevant tickets?
+
+<!-- Link issues here -->
+
+### Where should the reviewer start?
```
[the remark this answers is not in the document]
Comment (Aschen) [reply .github/pull_request_template.md]: Yes I agree, maybe I can add a comment to tell this to the creator

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -2,34 +2,31 @@
   This template is not mandatory.
   It simply serves to provide a guide to allow a better review of pull requests.
 -->
-### What's is in this PR ?
+
+## What is in this PR?
```
Comment (scottinet) [inline .github/pull_request_template.md]: I'm sorry, but now that I've given this some thoughts, I am against that section too: we already have that information with the `changelog:*` labels. 
So not only does this section duplicates that information, but our labels have been set using standard changelog categories, and are used by our release tool to generate a standardized version's changelog automatically.

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
+<!--
+  This template is not mandatory.
+  It simply serves to provide a guide to allow a better review of pull requests.
+-->
+
+## What is in this PR?
```
Comment (scottinet) [inline .github/pull_request_template.md]: I'm sorry, but now that I've given this some thoughts, I am against that section too: we already have that information with the `changelog:*` labels. 
So not only does this section duplicates that information, but our labels have been set using standard changelog categories, and are used by our release tool to generate a standardized version's changelog automatically.

Comment (scottinet) [reply .github/pull_request_template.md]: Following our discussion: this section can be replaced by a comment telling that putting a `changelog` label is strongly encouraged (PRs without such a label will be shown in the `Others` section of generated changelogs)

**Comment by Aschen** [comment]
For kuzzleio/kuzzle we have labels but I aim to include this template in every kuzzleio repository.  
So for this one we can remove the `What's is in this PR ?` section but for the other repository it would be great to keep it I think.

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -3,20 +3,12 @@
...
-<!-- Add a link for each related issues (eg: [X] Bug fix #42 #21 #84 ) -->
-
-
-- [ ] Bug fix <!-- non-breaking change which fixes an issue -->
-- [ ] New feature <!-- non-breaking change which adds functionality) -->
-- [ ] Breaking change <!-- fix or feature that would cause existing functionality to not work as expected) -->
-
+<!--
+  IMPORTANT
+  Don't forget to add the corresponding label to your PR.
```
Comment (scottinet) [inline .github/pull_request_template.md]: (nitpicking) I think that the following is clearer: `on't forget to add the corresponding "changelog:xxx" label to your PR.`

**Pull request merged**